msticpy/context/azure/sentinel_core.py

```diff
@@ -199,57 +283,63 @@ def connect(
                 "and specify the new cloud name using the `cloud` parameter.",
                 title="Cannot switch cloud at connect time",
             )
-        tenant_id = tenant_id or self.workspace_config[WorkspaceConfig.CONF_TENANT_ID]
         logger.info("Using tenant id %s", tenant_id)
-        self._token = kwargs.pop("token", None)
-        super().connect(
-            auth_methods=auth_methods, tenant_id=tenant_id, silent=silent, **kwargs
-        )
+        az_connect_kwargs = {
+            key: value
+            for key, value in connect_kwargs.items()
+            if key not in _WS_PARAMETERS
+        }
+        if tenant_id:
+            az_connect_kwargs["tenant_id"] = tenant_id
+        self._token = az_connect_kwargs.pop("token", None)
+        super().connect(auth_methods=auth_methods, silent=silent, **az_connect_kwargs)
         if not self._token:
             logger.info("Getting token for %s", tenant_id)
             self._token = get_token(
                 self.credentials, tenant_id=tenant_id, cloud=self.cloud  # type: ignore
             )
 
-        with contextlib.suppress(KeyError):
-            logger.info(
-                "Setting default subscription to %s from workspace settings",
-                self.default_subscription,
-            )
-            self.default_subscription = self.workspace_config[
-                WorkspaceConfig.CONF_SUB_ID_KEY
-            ]
-            self.set_default_workspace(
-                self.default_subscription,
-                self.workspace_config[WorkspaceConfig.CONF_WS_NAME_KEY],
-            )
-        self._create_api_paths_for_workspace(
-            az_resource_id=None,
-            subscription_id=self.workspace_config.get(WorkspaceConfig.CONF_SUB_ID_KEY),
-            resource_group=self.workspace_config.get(
-                WorkspaceConfig.CONF_RES_GROUP_KEY
-            ),
-            workspace_name=self.workspace_config.get(WorkspaceConfig.CONF_WS_NAME_KEY),
-        )
-
     def _create_api_paths_for_workspace(
         self,
         az_resource_id: Optional[str] = None,
-        subscription_id: Optional[str] = None,
-        resource_group: Optional[str] = None,
-        workspace_name: Optional[str] = None,
+        subscription_id: str = "",
```

Comment:
I think these params and the arguments to `self._build_sent_res_id` should just take an instance of `sentinel_instance_details: SentinelInstanceDetails`?

Reply:
agreed.